custom-remote-user-store-manager/manager/src/main/java/org/wso2/carbon/remote/user/store/manager/WSUserStoreManager.java

```diff
@@ -0,0 +1,388 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.remote.user.store.manager;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.remote.user.store.manager.util.EndpointUtil;
+import org.wso2.carbon.user.api.Properties;
+import org.wso2.carbon.user.api.Property;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.claim.ClaimManager;
+import org.wso2.carbon.user.core.common.AbstractUserStoreManager;
+import org.wso2.carbon.user.core.common.RoleContext;
+import org.wso2.carbon.user.core.profile.ProfileConfigurationManager;
+import org.wso2.carbon.user.core.tenant.Tenant;
+import org.wso2.carbon.user.core.util.JDBCRealmUtil;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * User store manager for the web service user store manager.
+ */
+public class WSUserStoreManager extends AbstractUserStoreManager {
+
+    private static final String WS_UM_PROPERTY_URL_NAME = "url";
+    private static final String WS_UM_PROPERTY_URL_DISPLAY_NAME = "http://localhost:8080/sample-service";
+    private static final String WS_UM_PROPERTY_URL_DESCRIPTION = "Endpoint URL for the user store web service";
+    private static Log log = LogFactory.getLog(WSUserStoreManager.class);
+    private CloseableHttpClient closeableHttpClient;
+
+    public WSUserStoreManager() {
+
+    }
+
+    public WSUserStoreManager(RealmConfiguration realmConfig, Map<String, Object> properties,
+                              ClaimManager claimManager, ProfileConfigurationManager profileManager, UserRealm realm,
+                              Integer tenantId) throws UserStoreException {
+
+        this.realmConfig = realmConfig;
+        this.claimManager = claimManager;
+        this.userRealm = realm;
+        this.tenantId = tenantId;
+        realmConfig.setUserStoreProperties(JDBCRealmUtil.getSQL(realmConfig
+                .getUserStoreProperties()));
```

Comment:
Formatting issue.

Reply:
Fixed in 09486ce0c6187db78c1b97e4eaefd890c511141b.